spec/find-view-spec.coffee

```diff
@@ -892,6 +956,39 @@ describe 'FindView', ->
         findView.caseOptionButton.click()
         expect(editor.getSelectedBufferRange()).toEqual [[2, 0], [2, 5]]
 
+      describe "when there are existing selections", ->
+        describe "when any existing selections match the pattern", ->
+          it "shouldn't jump to the next match when toggled on", ->
+            findView.model.setFindOptions caseSensitive: false
+            findView.findEditor.setText 'WORDs'
+            editor.setSelectedBufferRange [[2, 0], [2, 5]]
+            findView.caseOptionButton.click()
+            expect(editor.getSelectedBufferRange()).toEqual [[2, 0], [2, 5]]
+
+          it "shouldn't jump to the next match when toggled off", ->
+            findView.model.setFindOptions caseSensitive: true
+            findView.findEditor.setText 'WORDs'
+            editor.setSelectedBufferRange [[2, 0], [2, 5]]
+            findView.caseOptionButton.click()
+            expect(editor.getSelectedBufferRange()).toEqual [[2, 0], [2, 5]]
+
+        describe "when no existing selections match the pattern", ->
+          it "should jump to the next match when toggled on", ->
+            findView.model.setFindOptions caseSensitive: false
+            findView.findEditor.setText 'WORDs'
+            editor.setSelectedBufferRange [[1, 0], [1, 5]]
+            findView.caseOptionButton.click()
+            expect(editor.getSelectedBufferRange()).toEqual [[2, 0], [2, 5]]
+
+          it "should jump to the next match when toggled off", ->
+            # It's impossible to create a test where `caseSensitive: true` matches but `caseSensitive: false` doesn't.
+            # Instead we'll create a generic selection and ensure that it jumps to the next match.
+            findView.model.setFindOptions caseSensitive: true
+            findView.findEditor.setText 'WORDs'
+            editor.setSelectedBufferRange [[0, 0], [0, 5]]
+            findView.caseOptionButton.click()
+            expect(editor.getSelectedBufferRange()).toEqual [[1, 0], [1, 5]]
+
```

Comment:
I usually don't say this, but I think there might be a few *too* many tests for this code change. They look very good, but I would probably combine the tests for `when toggled on` and `when toggled off` into one, in which you click the button twice, and check that the selection persists at every stage.